packages/legend-studio/src/stores/EditorStore.ts

```diff
@@ -790,24 +791,39 @@ export class EditorStore {
     this.initState.setMessage(`Fetching entities...`);
     try {
       // fetch workspace entities and config at the same time
+      const projectId = this.sdlcState.activeProject.projectId;
+      const activeWorkspace = this.sdlcState.activeWorkspace;
       const result = (yield Promise.all([
-        this.sdlcServerClient.getEntities(
-          this.sdlcState.activeProject.projectId,
-          this.sdlcState.activeWorkspace,
-        ),
-        this.sdlcServerClient.getConfiguration(
-          this.sdlcState.activeProject.projectId,
-          this.sdlcState.activeWorkspace,
-        ),
+        this.sdlcServerClient.getEntities(projectId, activeWorkspace),
+        this.sdlcServerClient.getConfiguration(projectId, activeWorkspace),
       ])) as [Entity[], PlainObject<ProjectConfiguration>];
       entities = result[0];
       projectConfiguration = result[1];
+
+      const projectConfigurationPayload = projectConfiguration;
+      const projectDependencyList =
+        projectConfiguration.projectDependencies as {
+          projectId: string;
+          versionId: string;
+        }[];
+      projectConfigurationPayload.projectDependencies =
+        projectDependencyList.map(
```

Comment:
you shouldn't need to do this. you should be able to just call `fromJSON` on this.